src/main/database.cpp

```diff
@@ -50,6 +65,7 @@ Database::Database(std::string databasePath, SystemConfig systemConfig)
     bufferManager = std::make_unique<BufferManager>(this->systemConfig.bufferPoolSize);
     memoryManager = std::make_unique<MemoryManager>(bufferManager.get());
     queryProcessor = std::make_unique<processor::QueryProcessor>(this->systemConfig.maxNumThreads);
+    checkAccessMode();
```

Comment:
Can we move this function inside `initDBDirAndCoreFilesIfNecessary`? You can check access mode after 
```cpp
    if (!FileUtils::fileOrPathExists(databasePath)) {
        FileUtils::createDir(databasePath);
    }
```